staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/cache.go

```diff
@@ -55,5 +66,25 @@ func (c *simpleCache) set(key []byte, transformer value.Transformer) {
 	if transformer == nil {
 		panic("transformer must not be nil")
 	}
-	c.cache.Set(base64.StdEncoding.EncodeToString(key), transformer, c.ttl)
+	c.cache.Set(c.keyFunc(key), transformer, c.ttl)
+}
+
+// keyFunc generates a string key by hashing the inputs.
+// This lowers the memory requirement of the cache.
+func (c *simpleCache) keyFunc(s []byte) string {
+	h := c.hashPool.Get().(hash.Hash)
+	h.Reset()
+
+	key := toString(h.Sum(nil)) // skip base64 encoding to save an allocation
+	if _, err := h.Write(s); err != nil {
```

Comment:
Write has to come before Sum.

Reply:
thanks for catching it! i was just validating that locally too. It should be updated now.